Core/Src/main.c

```diff
@@ -87,17 +98,47 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // SPI
+  SPI_HandleTypeDef *pSPI_Handle = &hspi1;
+  uint8_t *TxBuffer = malloc(BYTE_COUNT * sizeof(uint8_t));
+  uint8_t *RxBuffer = malloc(BYTE_COUNT * sizeof(uint8_t));
+
+  TxBuffer[0] = 0x01;
+  TxBuffer[1] = 0x80;
+  TxBuffer[2] = 0x00;
+
+  uint16_t RxSegOne = 0;
+  uint16_t RxComplete = 0;
+
+  // PWM
+  TIM_HandleTypeDef *pTIM_Handler = &htim1;
+  HAL_TIM_PWM_Start(pTIM_Handler, 1);
+
+  uint16_t compRegVal = 0;
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
-    /* USER CODE END WHILE */
+	  if(HAL_SPI_TransmitReceive(pSPI_Handle, TxBuffer, RxBuffer, BYTE_COUNT, SPI_TIMEOUT) == HAL_OK)
+	  {
```

Comment:
function looks good, but the CS line was never set LOW. This means that the peripheral ADC would never start taking in data.